src/Tribe/RSVP.php

```diff
@@ -915,139 +876,94 @@ public function create_attendee_for_ticket( $ticket, $attendee_data ) {
-		 * @param int $post_id           Event post ID.
-		 * @param int $product_id        RSVP ticket post ID.
-		 * @param int $order_attendee_id Attendee # for order.
-		 */
-		do_action( 'event_tickets_rsvp_ticket_created', $attendee_id, $post_id, $product_id, $order_attendee_id );
-
-		if ( null === $order_attendee_id ) {
-			/**
-			 * Action fired when an RSVP ticket has had attendee tickets generated for it.
-			 *
-			 * @param int    $product_id RSVP ticket post ID.
-			 * @param string $order_id   ID (hash) of the RSVP order.
-			 * @param int    $qty        Quantity ordered.
-			 */
-			do_action( 'event_tickets_rsvp_tickets_generated_for_product', $product_id, $order_id, 1 );
-
-			$this->clear_attendees_cache( $post_id );
-		}
-
 		return $attendee_id;
 	}
 
 	/**
 	 * Update the RSVP values for this user.
 	 *
-	 * Note that, within this method, $order_id refers to the attendee or ticket ID
+	 * Note that, within this method, $attendee_id refers to the attendee or ticket ID
 	 * (it does not refer to an "order" in the sense of a transaction that may include
 	 * multiple tickets, as is the case in some other methods for instance).
 	 *
-	 * @param array $data
-	 * @param int   $order_id
-	 * @param int   $event_id
+	 * @param array $attendee_data Information that we are trying to save.
+	 * @param int   $attendee_id   The attendee ID.
+	 * @param int   $post_id       The event/post ID.
 	 */
-	public function update_attendee_data( $data, $order_id, $event_id ) {
+	public function update_attendee_data( $attendee_data, $attendee_id, $post_id ) {
 		$user_id = get_current_user_id();
 
-		$rsvp_orders    = $this->tickets_view->get_event_rsvp_attendees( $event_id, $user_id );
-		$rsvp_order_ids = array_map( 'absint', wp_list_pluck( $rsvp_orders, 'order_id' ) );
+		$rsvp_attendees    = $this->tickets_view->get_event_rsvp_attendees( $post_id, $user_id );
+		$rsvp_attendee_ids = array_map( 'absint', wp_list_pluck( $rsvp_attendees, 'attendee_id' ) );
 
 		// This makes sure we don't save attendees for orders that are not from this current user and event
-		if ( ! in_array( (int) $order_id, $rsvp_order_ids, true ) ) {
+		if ( ! in_array( $attendee_id, $rsvp_attendee_ids, true ) ) {
 			return;
 		}
 
 		$attendee = array();
 
-		// Get the Attendee Data, it's important for testing
-		foreach ( $rsvp_orders as $test_attendee ) {
-			if ( $order_id !== $test_attendee['order_id'] ) {
+		// Get the attendee data for the attendee we are updating.
+		foreach ( $rsvp_attendees as $test_attendee ) {
+			if ( $attendee_id !== $test_attendee['attendee_id'] ) {
 				continue;
 			}
 
 			$attendee = $test_attendee;
 		}
 
 		// Don't try to Save if it's restricted
-		if ( ! isset( $attendee['product_id'] )
-		     || $this->tickets_view->is_rsvp_restricted( $event_id, $attendee['product_id'] )
+		if (
+			! isset( $attendee['product_id'] )
+		    || $this->tickets_view->is_rsvp_restricted( $post_id, $attendee['product_id'] )
```

Comment:
:no_entry_sign: Tabs must be used to indent lines; spaces are not allowed

<sub>hash: 20054402bfdc89d63b06ae3f500f5ea8</sub>